### PR TITLE
remote haulers and links fix

### PR DESCRIPTION
**Pull request by Vykook**
this fix prevents hauler from another unload destination, if he alredy
started unloading to link.

### CODE REVIEW

**Comment by Vykook** [comment]
fixing #1178 PR.

File: creep.behaviour.remoteHauler.js [hunk 1]
```diff
@@ -27,9 +27,11 @@ mod.nextAction = function(creep){
             if( creep.room.storage ) deposit.push(creep.room.storage);
             // containers?
             if( creep.room.structures.container ) deposit = deposit.concat( creep.room.structures.container.privateers );
+            if( creep.memory.lastLink ) deposit.push(Game.getObjectById(creep.memory.lastLink));
```
Comment (Spedwards) [inline creep.behaviour.remoteHauler.js]: Move this below the `deposit = links` line.

Comment (Spedwards) [inline creep.behaviour.remoteHauler.js]: By what you've put in the PR description, it sounds like all you really need is the following:

```
if (creep.data.lastLink) {
    let target = Game.getObjectById(creep.data.lastLink);
    if (target && target.pos.inRangeTo(creep.pos, 2) && this.assign(creep, Creep.action.charging, target)) {
        return;
    } else {
        delete creep.data.lastLink;
    }
}
```

Then you could remove this line and line 49.

That will avoid the `findClosestByRange` call and should also serve your purpose.

**Comment by karlthepagan** [comment]
If these changes are addressed I would consider this for 0.7

Link performance at RCL5-6 needs this improvement.

**Comment by chobobobo** [comment]
@Vykook please address the requested changes so that this can be tested and merged

**Pull request closed without merging**